src/com/griddynamics/devops/mpl/Helper.groovy

```diff
@@ -126,19 +126,51 @@ abstract class Helper {
   }
 
   /**
-   * Cutting a stacktrace to just first execution of the module and one before
+   * Cut & simplify a stacktrace
    *
    * @param exception  container of the stacktrace
+   *
    * @return  List with stack trace elements
    */
   static StackTraceElement[] getModuleStack(Throwable exception) {
     List stack = exception.getStackTrace()
+
+    // For jenkins to remove starting trace items
+    if( stack.last()?.getFileName() == 'Thread.java' ) {
+      // Finding the first MPLModule call and cutting the trace
+      for( def i = stack.size(); i--; i > 0 ) {
+        if( stack[i-1].getFileName()?.contains('MPLModule.groovy') )
```

Comment:
In the end of loop you will remove first frame if last (non-removed) is MPLModule.groovy.

Reply:
So, the for(;;) logic is simple:
1. Execute first statement
2. Check bool(second statement) or exit
3. Execute third statement
4. Execute body
5. goto 2

```groovy
println "Forward"
for( def i=0; i<10; i++ )
  println i

println "Reverse"
for( def i=10; i>0; i-- )
  println i

println "Not typical forward"
for( def i=0; i++; )
  println i
println "Why not working: ${0 as Boolean}"

println "Not typical reverse"
for( def i=10; i--; )
  println i
```

Check the difference.